core/cas-server-core-web/src/main/java/org/apereo/cas/config/CasCoreWebConfiguration.java

```diff
@@ -80,4 +83,12 @@ public ArgumentExtractor argumentExtractor(final List<ServiceFactoryConfigurer>
         configurers.forEach(c -> serviceFactoryList.addAll(c.buildServiceFactories()));
         return new DefaultArgumentExtractor(serviceFactoryList);
     }
+
+    @Bean
+    public FactoryBean<UrlValidator> urlValidator() {
+        final SimpleUrlValidatorFactoryBean bean = new SimpleUrlValidatorFactoryBean();
+        bean.setCasProperties(this.casProperties);
```

Comment:
Also it would be best if you could use constructor injection rather than a setter.